src/app/vendors/vendor-item.ts

```diff
@@ -31,6 +32,29 @@ export interface VendorItem {
   readonly displayCategoryIndex?: number;
   readonly costs: DestinyItemQuantity[];
   readonly previewVendorHash?: number;
+  /** The state of this item in the user's D2 Collection */
+  readonly collectibleState?: DestinyCollectibleState;
+}
+
+/**
+ * Find the state of this item in the user's collections. This takes into account
+ * the selected character.
+ */
+function getCollectibleState(
+  inventoryItem: DestinyInventoryItemDefinition,
+  profileResponse: DestinyProfileResponse | undefined,
+  characterId: string
+) {
+  const collectibleHash = inventoryItem.collectibleHash;
+  let collectibleState: DestinyCollectibleState | undefined;
+  if (collectibleHash) {
+    collectibleState =
+      profileResponse?.profileCollectibles?.data?.[collectibleHash]?.state ??
+      (characterId
+        ? profileResponse?.characterCollectibles?.data?.[characterId]?.[collectibleHash]?.state
```

Comment:
I'm not getting any blue "unlocked in collections" checkmarks with this, because `(profile|character)Collectibles?.data` isn't meant to be indexed, this should be `(profile|character)Collectibles?.data?.collectibles`.

Reply:
It looks like thr TS compiler option `"suppressImplicitAnyIndexErrors": false,` would catch this and triggers only on a small number of cases in DIM (mostly d2ai JSON).

Reply:
Hm, I was using the "only show uncollected items" feature to test this, I wonder why it still worked. For removing `suppressImplicitAnyIndexErrors`, I have tried it occasionally, but there's a *lot* of placed that need to be changed.